app/services/record_declarations/base.rb

```diff
@@ -82,13 +97,15 @@ def declaration_parameters
     end
 
     def record_exists_with_different_declaration_date?
-      declaration = self.class.declaration_model.find_by(
-        user: user,
-        course_identifier: course_identifier,
-        declaration_type: declaration_type,
-      )
-
-      declaration.present? && !declaration.voided? && declaration.declaration_date != Time.zone.parse(declaration_date)
+      declaration = self.class.declaration_model
+                      .where.not(state: self.class.declaration_model.states[:voided])
```

Comment:
What's the advantage of this vs below?

```suggestion
                      .where.not(state: ParticipantDeclaration.states[:voided])
```